api/src/test/java/com/epam/pipeline/acl/issue/IssueApiServiceTest.java

```diff
@@ -0,0 +1,329 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.acl.issue;
+
+import com.epam.pipeline.controller.vo.EntityVO;
+import com.epam.pipeline.controller.vo.IssueCommentVO;
+import com.epam.pipeline.controller.vo.IssueVO;
+import com.epam.pipeline.entity.datastorage.aws.S3bucketDataStorage;
+import com.epam.pipeline.entity.issue.Issue;
+import com.epam.pipeline.entity.issue.IssueComment;
+import com.epam.pipeline.entity.security.acl.AclClass;
+import com.epam.pipeline.manager.EntityManager;
+import com.epam.pipeline.manager.issue.IssueManager;
+import com.epam.pipeline.manager.security.AuthManager;
+import com.epam.pipeline.security.acl.AclPermission;
+import com.epam.pipeline.test.acl.AbstractAclTest;
+import com.epam.pipeline.test.creator.datastorage.DatastorageCreatorUtils;
+import com.epam.pipeline.test.creator.issue.IssueCreatorUtils;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+public class IssueApiServiceTest extends AbstractAclTest {
+
+    private final EntityVO entityVO = IssueCreatorUtils.getEntityVO(ID, AclClass.DATA_STORAGE);
+    private final Issue issue = IssueCreatorUtils.getIssue(entityVO, SIMPLE_USER);
+    private final IssueVO issueVO = IssueCreatorUtils.getIssueVO(entityVO);
+    private final S3bucketDataStorage s3bucket = DatastorageCreatorUtils.getS3bucketDataStorage(ID, OWNER_USER);
+    private final List<Issue> issueList = Collections.singletonList(issue);
+    private final IssueComment issueComment = IssueCreatorUtils.getIssueComment(SIMPLE_USER);
+    private final IssueCommentVO issueCommentVO = IssueCreatorUtils.getIssueCommentVO();
+
+    @Autowired
+    private IssueApiService issueApiService;
+
+    @Autowired
+    private IssueManager mockIssueManager;
+
+    @Autowired
+    private EntityManager mockEntityManager;
+
+    @Autowired
+    private AuthManager mockAuthManager;
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldCreateIssueForAdmin() {
+        doReturn(issue).when(mockIssueManager).createIssue(issueVO);
+
+        assertThat(issueApiService.createIssue(issueVO)).isEqualTo(issue);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldCreateIssueWhenPermissionIsGranted() {
+        initAclEntity(s3bucket, AclPermission.READ);
+        doReturn(issue).when(mockIssueManager).createIssue(issueVO);
+        doReturn(s3bucket).when(mockEntityManager).load(AclClass.DATA_STORAGE, ID);
```

Comment:
Similarly let's extract the acl class to a field and refer to it as just `ACL_CLASS` or `ENTITY_ACL_CLASS`.